### PR TITLE
add Invert filter and Outline filter

### PR DESCRIPTION
**Pull request by Delta0533**
Invert filter formula :
Inverted Color = 255 − Original Color

Outline filter :
step1 - Convert image to grayscale for Easy to edge detection (reduce complexity from 3 color channel(RGB) to 1 channel(grayscale))

step2 - define sobel kernel in horizortal and vertical edges

step3 - matrix convolution between sobel kernel and image

step4 - calculate threshold value (for separate edge and non-edge) and convert image back to RGB channel

---------------------------------------------------------------------------------------------------------------------------------------------

This pull request introduces two new filter types, "Invert" and "Outline," to the image filtering functionality. The changes include updates to the filter selection UI, the filter type enumeration, and the filter application logic, along with the addition of a new function for the outline effect.

### Enhancements to Image Filtering:

* **UI Update for Filter Selection**:
  - Added "Invert" and "Outline" options to the filter dropdown menu in `src/main.py`. (`src/main.py`)

* **Filter Type Enumeration**:
  - Added `INVERT` and `OUTLINE` to the `FilterType` enumeration in `src/tools/image_filter_color.py`. (`src/tools/image_filter_color.py`)

* **Filter Settings**:
  - Defined settings for `INVERT` and `OUTLINE` filter types. (`src/tools/image_filter_color.py`)

* **Filter Application Logic**:
  - Updated `apply_filter` function to handle `INVERT` and `OUTLINE` filter types. (`src/tools/image_filter_color.py`)

* **Outline Effect Implementation**:
  - Added `apply_outline` function to implement the outline effect using Sobel edge detection. (`src/tools/image_filter_color.py`)

### CODE REVIEW

File: src/main.py
Comment (paratpanu18) [inline src/main.py:1]: This part of code seems not scalable, since when adding new filter type, we always have to edit this manually. Best practice is to use python unpacking from `FilterType` enum items.
```python
filter_list = [filter.value for filter in FilterType]
filter_menu = ttk.OptionMenu(frame, filter_option, "None", "None", *filter_list)
```

File: src/tools/image_filter_color.py [hunk 1]
```diff
@@ -1,12 +1,16 @@
 from enum import Enum
 import numpy as np
 import cv2
+from tools.grayscale import rgb_to_grayscale
+from tools.blur import apply_blur
```
Comment (paratpanu18) [inline src/tools/image_filter_color.py:5]: What is this function used for?

File: src/tools/image_filter_color.py [hunk 2]
```diff
@@ -39,11 +51,76 @@ def apply_filter(image: np.ndarray, filter_type: FilterType) -> np.ndarray:
...
 
+def apply_outline(image: np.ndarray) -> np.ndarray:
+    """
+    Apply an outline effect to the image using a Sobel filter for edge detection.
+    
+    :param image: Input image in BGR format (height, width, 3).
+    :return: Image with outline effect applied.
+    """
+
+    gray = rgb_to_grayscale(image)
```
Comment (paratpanu18) [inline src/tools/image_filter_color.py:75]: Varible's name should be more clear.

**Review by paratpanu18 — COMMENTED** [review]
I've left a few comments for you.

File: src/main.py
Comment (paratpanu18) [reply src/main.py:1]: By the way, I've solved this problem in my own branch which should be okay when merging with your. But if you want to know how this works you can visit: https://geekflare.com/python-unpacking-operators/

TLDR: You have to do nothing with this

File: src/tools/image_filter_color.py (hunk 1 above)
Comment (Delta0533) [reply src/tools/image_filter_color.py:5]: don't use blur function I forgot to remove this.

**Comment by paratpanu18** [comment]
LGTM
